### PR TITLE
Added symmetric SpMV implementation

### PR DESCRIPTION
**Pull request by athelaf**

### CODE REVIEW

File: spmv/Matrix.cpp [hunk 1]
```diff
@@ -29,6 +82,15 @@ Matrix<T>::~Matrix()
 #endif
 }
 
+template <typename T>
+size_t Matrix<T>::size() const
```
Comment (IgorBaratta) [inline spmv/Matrix.cpp:86]: We should probably add some documentation... 
I wouldn't expect this function to return the size in bytes .

File: spmv/Matrix.h [hunk 2]
```diff
@@ -30,10 +30,27 @@ class Matrix
...
 
   /// Destructor (destroys MKL structs, if using MKL)
   ~Matrix();
 
+  // Number of rows in the matrix
+  int rows() const { return _matA.rows(); }
+  // Number of columns in the matrix
+  int cols() const { return _matA.cols(); }
+  // Number of nonzeros in the matrix
+  int nonZeros() const { return (_symmetric) ? _matA.nonZeros() : _nnz; }
```
Comment (IgorBaratta) [inline spmv/Matrix.h:49]: I think we are using `snake_case` convention. ` non_zeros` maybe?

**Review by IgorBaratta — APPROVED** [review]
Seems good.
Just some minor comments.

**Comment by athelaf** [comment]
Can we merge this branch or is there anything else required? Other work I have done builds upon this one...

**Review by IgorBaratta — APPROVED** [review]

**Pull request merged**